test/riak_kv_sweeper_SUITE.erl

```diff
@@ -0,0 +1,892 @@
+                                       end),
+
+    ok = receive_msg({ok, successful_sweep, sweeper_callback_wait, WaitIndex}),
+    tick_per_index(Indices),
+    sweeper_callback_wait:before_visit([WaitIndex1]),
+    ok = receive_msg({ok, successful_sweep, sweeper_callback_wait, WaitIndex1}),
+    ok.
+
+test_scheduler_sweep_window_never(Config) ->
+    Indices = ?config(vnode_indices, Config),
+    new_sweep_participant(sweeper_callback_1),
+    sweeper_meck_helper:backend_modules(async, _NumKeys=5000, _ObjecSizeBytes=0),
+
+    application:set_env(riak_kv, sweep_window, never),
+    riak_kv_sweeper:enable_sweep_scheduling(),
+    StatusBefore = riak_kv_sweeper:status(),
+    riak_kv_sweeper:sweep_tick(),
+    StatusAfter = riak_kv_sweeper:status(),
+    StatusAfter = StatusBefore,
+    application:set_env(riak_kv, sweep_window, always),
+    sweep_all_indices(Indices),
+    ok.
+
+test_scheduler_now_outside_sleep_window(Config) ->
+    Indices = ?config(vnode_indices, Config),
+    new_sweep_participant(sweeper_callback_1),
+    sweeper_meck_helper:backend_modules(async, _NumKeys=5000, _ObjecSizeBytes=0),
+
+    {_, {Hour, _, _}} = calendar:local_time(),
+    Start = add_hours(Hour, 2),
+    End = add_hours(Start, 1),
+    application:set_env(riak_kv, sweep_window, {Start, End}),
+    StatusBefore = riak_kv_sweeper:status(),
+    riak_kv_sweeper:sweep_tick(),
+    StatusAfter = riak_kv_sweeper:status(),
+    StatusAfter = StatusBefore,
+    application:set_env(riak_kv, sweep_window, {Hour, add_hours(Hour, 1)}),
+    sweep_all_indices(Indices),
+    ok.
+
+%%
+%% This test runs long ~5 seconds because the it schedulers `expired'
+%% and the scheduler has a minimum resolution of 1 second in this
+%% case.
+test_stop_all_scheduled_sweeps(Config) ->
+    Indices = ?config(vnode_indices, Config),
+    new_sweep_participant(sweeper_callback_wait),
+    sweeper_meck_helper:backend_modules(async, _NumKeys=5000, _ObjecSizeBytes=0),
+
+    application:set_env(riak_kv, sweep_concurrency, length(Indices)),
+    riak_kv_sweeper:enable_sweep_scheduling(),
+    %% Lets sweeps run once so they are not in never run state
+    tick_per_index(Indices),
+    sweeper_callback_wait:before_visit(Indices),
+    [ok = receive_msg({ok, successful_sweep, sweeper_callback_wait, I}) || I <- Indices],
+    riak_kv_sweeper:sweep_tick(),
+    {_, Sweeps} = riak_kv_sweeper:status(),
+    sweeper_callback_wait:before_visit(any, fun(_) ->
+                                                    Running = riak_kv_sweeper:stop_all_sweeps(),
+                                                    true = Running >= 1
+                                            end),
+    Running = [riak_kv_sweeper_state:sweep_index(Sweep) ||
+                  Sweep <- Sweeps,
+                  riak_kv_sweeper_state:sweep_state(Sweep) == running],
+    [ok = receive_msg({ok, failed_sweep, sweeper_callback_wait, I}) || I <- Running],
+    ok = check_sweeps_idle(Indices),
+    ok.
```

Comment:
We could eliminate this extra line, since the previous expression already evaluates to `ok`.